controllers/cbcontainersagent_controller.go

```diff
@@ -161,9 +181,20 @@ func (r *CBContainersAgentController) getAccessToken(ctx context.Context, cbCont
 	return accessToken, nil
 }
 
+func (r *CBContainersAgentController) updateCRStatus(ctx context.Context, cbContainersCluster *cbcontainersv1.CBContainersAgent, agentStateWasChanged bool) error {
+	// If we don't expect more changes (i.e. nothing changed in reality) and we haven't updated the status, we do so now.
+	if !agentStateWasChanged && cbContainersCluster.Status.ObservedGeneration < cbContainersCluster.ObjectMeta.Generation {
+		r.Log.Info("Updating CBContainersAgent status")
+		cbContainersCluster.Status.ObservedGeneration = cbContainersCluster.ObjectMeta.Generation
+		return r.Client.Status().Update(ctx, cbContainersCluster)
+	}
+	return nil
+}
+
 func (r *CBContainersAgentController) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&cbcontainersv1.CBContainersAgent{}).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
```

Comment:
This filter means changes to the CR's `status` and `metadata` will not trigger reconcile loops. I think that is OK and it saves unnecessary reconciles.

Note that we still get reconcile loops if any of the owned resources changes.